allennlp/data/fields/adjacency_field.py

```diff
@@ -90,9 +90,8 @@ def count_vocab_items(self, counter: Dict[str, Dict[str, int]]):
 
     @overrides
     def index(self, vocab: Vocabulary):
-        if self._indexed_labels is None and self.labels is not None:
```

Comment:
I'm actually not sure about this - @DeNeutoy, you know more about what this field does.  It looks like I should probably keep a check here for `self.labels is not None`?